pkg/metrics/series.go

```diff
@@ -111,6 +118,45 @@ func (serie *Serie) hasDeviceTag() bool {
 	})
 }
 
+// PopulateResources removes any `dd.internal.resource:` tags in the series tags and uses the values to
+// populate the Serie.Resources field. The format for the dd.internal.resource tag values is
+// <resource_type>,<resource_name>. Any dd.internal.resource tag not matching the expected format
+// will be dropped.
+func (serie *Serie) PopulateResources() {
+	if !serie.hasResourceTag() {
+		return
+	}
+	// make a copy of the tags array. Otherwise the underlying array won't have
+	// the resource tag for the Nth iteration (N>1), and the resources field will
+	// be lost
+	filteredTags := make([]string, 0, serie.Tags.Len())
+
+	serie.Tags.ForEach(func(tag string) {
+		if strings.HasPrefix(tag, "dd.internal.resource:") {
```

Comment:
Would it make sense to make this a named constant? (Also use it instead of the hard-coded length down below).

Reply:
Totally, updated ✅ .